cmake/modules/unittest.cmake

```diff
@@ -33,6 +33,14 @@ if((NOT DEFINED ZEPHYR_BASE) AND (DEFINED ENV_ZEPHYR_BASE))
   set(ZEPHYR_BASE ${ENV_ZEPHYR_BASE} CACHE PATH "Zephyr base")
 endif()
 
+if(SOURCES)
+  message(WARNING
+      "Unit testing will soon stop automatically adding SOURCES to testbinary. If not already done,"
+      " please use:\n"
+      "    target_sources(testbinary PRIVATE \${SOURCES})"
+  )
```

Comment:
please move this to https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/FindDeprecated.cmake
along with this code:
https://github.com/zephyrproject-rtos/zephyr/blob/7ef05751a3f34030eb06dace23e357d10b33f460/cmake/modules/unittest.cmake#L36_L38
Plus a new `target_sources(testbinary PRIVATE ${SOURCES})` line, and then update code to use:

```
add_executable(testbinary)
find_package(Deprecated COMPONENTS SOURCES)
```

that allows us to have deprecated code located together in a single file and helps us to keep track and remove deprecated code after two release.

Plus, it has the benefit of already being able to clean up existing code :)

Note: align the warning message to similar messages in the deprecation module.